### DISCUSSION TITLE
Pitch shifting as an adjustable setting alongside offset

### DISCUSSION THREAD

**Discussion by Parakosmos** [ppy/osu#17883] [Ideas]
Often when I listen to music on youtube, I use a Chrome extension called Transpose to change the song's pitch. I don't know why I do this, but it gives me feelings of euphoria and allows me to harmonise myself with the music that's playing. I understand that Nightcore is a mod that increases the pitch of the played song, but I have always thought of being able to play pitch-adjusted music from any beatmap with any mod combination. 

I think the feature of **making pitch-shifting a local (beatmap-specific) or global audio setting** alongside offset and whatnot could enhance the osu! experience by broadening the ways you can listen to beatmaps. _By making it an audio setting players can choose to pitch shift without it affecting their leaderboard score submission (unlike how scores on osu!stable with mods like Relax don't save unless you decide to save it yourself), specifically on ranked & loved maps for competitve players ._ 

The feature could also be integrated as a player-specific trait of a replay where the replay's music could be played back in the pitch shift rate they used when playing the beatmap, which in my opinion, could further distinguish one player's replay from another just like with skins & hitsounds.

It would also be great to add a toggle option for Double-Time/Half-Time where the song's pitch is adjusted based on the speed of the song, just like how adjusting speed rate on Etterna changes the song's pitch using a toggled setting.

**Comment by ZenFlora** [comment]
If you play the game for the music, why are you requesting for a pitch-shifting mod outside of fun?

**Reply by Parakosmos** [reply]
I may have to edit that out because it's a confusing statement. I was trying to say that playing osu! is, for me, another way to experience music in general in both a casual (as getting invested in the song whilst playing) & competitve (playing the beatmap for high or desirable scores) way.

**Comment by frenzibyte** [comment]
> It would also be great to add a toggle option for Double-Time/Half-Time where the song's pitch is adjusted based on the speed of the song, just like how adjusting speed rate on Etterna changes the song's pitch using a toggled setting.

Isn't this basically `Nightcore`/`Daycore`? double-click `Double Time`/`Half Time` and you'll switch to those mods. Those mods pretty much change the pitch and the tempo of the beatmap at the same time, unlike DT/HT which only change the tempo.

I believe that might mark the main suggestion of adding a global pitch setting unnecessary. Since you'll still enhance the experience, via mods.

**Reply by Parakosmos** [reply]
> Isn't this basically Nightcore/Daycore? double-click Double Time/Half Time and you'll switch to those mods. Those mods pretty much change the pitch and the tempo of the beatmap at the same time, unlike DT/HT which only change the tempo.

I guess you have a point there with keeping the Nightcore/Daycore pitch changes as they are. They stick to a certain musical theme that require the pitches to match the tempo of the song. Didn't really take account the redunancy of using a global pitch changer there. 

But still for example, a lot of osu!mania beatmaps with rate modifiers have their pitches adjusted according to the tempo WITHOUT requiring Nightcore/Daycore & as I mentioned in Etterna you can choose for the tempo-pitch adjustment to be proportional or not. I don't think it would hurt for the latter feature to be considered as an adjustable and toggleable option on DT and HT mods if a global pitch changer setting is too ambitious of a suggestion.  

> I believe that might mark the main suggestion of adding a global pitch setting unnecessary. Since you'll still enhance the experience, via mods.

If you assume that most music listeners like pitch-adjusted audio either sped-up or sped-down, then yes the global setting could be considered redundant to add with the mods avaliable. However, I still see the potential of adding a pitch changer on non-NC/DC combinations (i.e. 1.0x tempo rate) as a feature that a small niche of players could enjoy if they are uncomfortable with using rate-adjusting modifications (e.g. newer, inexperienced players) or want to listen to a pitch-adjusted beatmap song on 1.0x rate (e.g. myself who likes to play HD).

**Reply by frenzibyte** [reply]
> I guess you have a point there with keeping the Nightcore/Daycore pitch changes as they are. They stick to a certain musical theme that require the pitches to match the tempo of the song. Didn't really take account the redunancy of using a global pitch changer there.

Keep in mind these mods may change to affect the pitch as well rather than leaving them at a constant state, see https://github.com/ppy/osu/discussions/13125

> If you assume that most music listeners like pitch-adjusted audio either sped-up or sped-down, then yes the global setting could be considered redundant to add with the mods avaliable. However, I still see the potential of adding a pitch changer on non-NC/DC combinations (i.e. 1.0x tempo rate) as a feature that a small niche of players could enjoy if they are uncomfortable with using rate-adjusting modifications (e.g. newer, inexperienced players) or want to listen to a pitch-adjusted beatmap song on 1.0x rate (e.g. myself who likes to play HD).

There's a separate discussion thread proposing a pitch-only mod adjustment, which should satisfy this (https://github.com/ppy/osu/discussions/13142).

You would usually want to listen to only a specific beatmap with a pitch in your mind, therefore a mod would be a better direction than a global setting (although nothing is decided here yet).

**Reply by Parakosmos** [reply]
> There's a separate discussion thread proposing a pitch-only mod adjustment, which should satisfy this (https://github.com/ppy/osu/discussions/13142).
You would usually want to listen to only a specific beatmap with a pitch in your mind, therefore a mod would be a better direction than a global setting (although nothing is decided here yet).

Ah cool so it's already had consideration in the past. Hopefully my post brought some kind of exposure back to the feature being possibly added.

**Reply by Parakosmos** [reply]
> Keep in mind these mods may change to affect the pitch as well rather than leaving them at a constant state, see https://github.com/ppy/osu/discussions/13125

That's very good to see! :D
